File: go/p2/p2.go
package main

import "fmt"

func main() {
	var ceiling uint
	ceiling = 4000000
	sum := fibonacciSequenceOfEvens(1, 2, ceiling)

	fmt.Println(sum)
}

func fibonacciSequenceOfEvens(y uint, x uint, ceiling uint) uint {
	var w, z uint
	z = 0

	for {
		w = y + x
		y = x
		x = w
		if y >= ceiling {
			break
		}
		z += isEven(y)
	}

	return z
}

func isEven(y uint) uint {
	if y%2 == 0 {
		return y
	}
	return 0
}

File: py/p2/p2.py
def fibonacci_sequence_of_evens(y, x, ceiling):
    z = 0
    while True:
        w = y + x
        y = x
        x = w

        if y >= ceiling:
            break

        z += is_even(y)
    return z

def is_even(x):
    if x % 2 == 0:
        return x
    return 0

if __name__ == '__main__':
    ceiling = 4000000
    sum = fibonacci_sequence_of_evens(1, 2, ceiling)
    print(sum)

